chips/lowrisc/src/hmac.rs

```diff
@@ -0,0 +1,248 @@
+//! HMAC (Hash-based Message Authentication Code).
+
+use core::cell::Cell;
+use kernel::common::cells::{OptionalCell, TakeCell};
+use kernel::common::registers::{
+    register_bitfields, register_structs, ReadOnly, ReadWrite, WriteOnly,
+};
+use kernel::common::StaticRef;
+use kernel::hil;
+use kernel::hil::hmac;
+use kernel::ReturnCode;
+
+register_structs! {
+    pub HmacRegisters {
+        (0x00 => intr_state: ReadWrite<u32, INTR_STATE::Register>),
+        (0x04 => intr_enable: ReadWrite<u32, INTR_ENABLE::Register>),
+        (0x08 => intr_test: ReadWrite<u32, INTR_TEST::Register>),
+        (0x0C => cfg: ReadWrite<u32, CFG::Register>),
+        (0x10 => cmd: ReadWrite<u32, CMD::Register>),
+        (0x14 => status: ReadOnly<u32, STATUS::Register>),
+        (0x18 => err_code: ReadOnly<u32>),
+        (0x1C => wipe_secret: WriteOnly<u32>),
+        (0x20 => key0: WriteOnly<u32>),
```

Comment:
Replace key0 - key7 to `[u32; 8]` instead?
The naming key0 - key8 doesn't help and it would simplify the code a bit i.e, indexing directly from an integer/loop.
What's the rationale behind this design?

The same applies for `digest0  - digest7`

Reply:
There wasn't really a rationale, just copying the registers from the documentation.

I'll convert it to an array.